app/tests/test_multiple_bucket_setup.py

```diff
@@ -0,0 +1,170 @@
+            UploadId="upload_id",
+            log_extra={
+                'parts': [],
+                'upload_id': 'upload_id',
+                'collection': self.asset.item.collection.name,
+                'item': self.asset.item.name,
+                'asset': self.asset.name
+            }
+        )
+
+    @parameterized.expand([
+        AVAILABLE_S3_BUCKETS.LEGACY,
+        AVAILABLE_S3_BUCKETS.MANAGED,
+    ],)
+    @patch("stac_api.s3_multipart_upload.MultipartUpload.call_s3_api")
+    def test_abort_multipart_upload(self, s3_bucket, mock_call):
+        executor = MultipartUpload(s3_bucket)
+
+        executor.abort_multipart_upload("foo", self.asset, "upload_id")
+
+        config = get_aws_settings(s3_bucket)
+
+        mock_call.assert_called_once_with(
+            executor.s3.abort_multipart_upload,
+            Bucket=config['STORAGE_BUCKET_NAME'],
+            Key="foo",
+            UploadId="upload_id",
+            log_extra={
+                'upload_id': 'upload_id', 'asset': self.asset.name
+            }
+        )
+
+    @parameterized.expand([
+        AVAILABLE_S3_BUCKETS.LEGACY,
+        AVAILABLE_S3_BUCKETS.MANAGED,
+    ],)
+    @patch("stac_api.s3_multipart_upload.MultipartUpload.call_s3_api")
+    def test_list_upload_parts(self, s3_bucket, mock_call):
+        executor = MultipartUpload(s3_bucket)
+
+        executor.list_upload_parts("foo", self.asset, "upload_id", 3, 2)
+
+        config = get_aws_settings(s3_bucket)
+
+        mock_call.assert_called_once_with(
+            executor.s3.list_parts,
+            Bucket=config['STORAGE_BUCKET_NAME'],
+            Key="foo",
+            UploadId="upload_id",
+            MaxParts=3,
+            PartNumberMarker=2,
+            log_extra={
+                'collection': self.asset.item.collection.name,
+                'item': self.asset.item.name,
+                'asset': self.asset.name,
+                'upload_id': 'upload_id',
+            }
+        )
+
+
+class TestBucketSelector(TestCase):
+
+    @parameterized.expand([('ch.meteoschweiz.ogd.rainfall', AVAILABLE_S3_BUCKETS.MANAGED),
+                           ('ch.meteoschweiz.ogd.downpour', AVAILABLE_S3_BUCKETS.MANAGED),
+                           ('ch.meteoschweiz.precipitation', AVAILABLE_S3_BUCKETS.LEGACY),
+                           ('ch.swisstopo.bgdi', AVAILABLE_S3_BUCKETS.MANAGED),
+                           ('ch.bafu.slipouts', AVAILABLE_S3_BUCKETS.LEGACY)])
+    def test_bucket_selection(self, collection_name, expected_bucket):
+        """Test if the pattern selection works"""
+        patterns = [r'ch\.meteoschweiz\.ogd.*', r'ch\.swisstopo\.bgdi.*']
+        with self.settings(MANAGED_BUCKET_COLLECTION_PATTERNS=patterns):
+            bucket_name = select_s3_bucket(collection_name)
+
+        self.assertEqual(bucket_name, expected_bucket)
+
+    @parameterized.expand([
+        'ch.meteoschweiz.ogd',
+        'ch.meteoschweiz.ogd.heatwave',
+        'ch.meteoschweiz.ogd.snowstorms',
```

Comment:
see comment above: `ch.meteoschweiz.ogd-*`

Reply:
This test becomes obsolete when we have the patterns in the environment.